src/sentry/replays/usecases/ingest/dom_index.py

```diff
@@ -300,27 +299,16 @@ def _should_report_hydration_error_issue(project_id: int) -> bool:
     ) and project.get_option("sentry:replay_hydration_error_issues")
 
 
-def _should_report_rage_click_issue(project_id: int) -> bool:
-    project = Project.objects.get(id=project_id)
-
-    def _project_has_feature_enabled() -> bool:
-        """
-        Check if the project has the feature flag enabled,
-        This is controlled by Sentry admins for release of the feature
-        """
-        return features.has(
-            "organizations:session-replay-rage-click-issue-creation",
-            project.organization,
-        )
-
-    def _project_has_option_enabled() -> bool:
-        """
-        Check if the project has the option enabled,
-        This is controlled by the project owner, and is a permanent setting
-        """
-        return project.get_option("sentry:replay_rage_click_issues")
-
-    return all([_project_has_feature_enabled(), _project_has_option_enabled()])
+def _should_report_rage_click_issue(project: Project) -> bool:
+    """
+    The feature is controlled by Sentry admins for release of the feature,
+    while the project option is controlled by the project owner, and is a
+    permanent setting
+    """
+    return features.has(
+        "organizations:session-replay-rage-click-issue-creation",
+        project.organization,
```

Comment:
What is the behavior of this relationship lookup?  Does it query the database?  Can we extract this out higher up so we're not querying in a loop?

Reply:
The handler queries options.. yes we can

Reply:
I'm talking about this dot access `project.organization`

Reply:
Ah I see. I think it's a DB lookup. I refactored it to be at the top level of the loop(s).